modules/ROOT/pages/authentication-authorization/ldap-integration.adoc

```diff
@@ -358,6 +359,105 @@ dbms.security.ldap.authorization.access_permitted_group=501
+
+[NOTE]
+====
+You need to change the `dbms.security.require_local_user` configuration setting to `true` to use Auth Objects. This means that a user with a matching Auth Object *must* exist in order to be able to authenticate and authorize.
+====
+
+The following examples show how you can use Cypher to do this.
+
+.Create a user with an Auth Object who can authenticate and authorize using `LDAP`.
+======
+----
+CREATE USER alice
+SET AUTH PROVIDER 'ldap' { SET ID 'cn=alice,ou=engineering,dc=example,dc=com' }
+----
+This will create a user who can authenticate and authorize using LDAP provided their LDAP `dn` is `cn=alice,ou=engineering,dc=example,dc=com`.
+
+When the `dbms.security.require_local_user` configuration setting is set to `true`, users can *only* authenticate when there is a user in the database with an Auth Object which links to the provider that the user is trying to authenticate with. If there is no matching Auth Object, the user will not be able to authenticate or authorize. This applies to all providers.
+======
+
+
+.Create a user with two Auth Objects allowing the user to authenticate and authorize with either LDAP or the `oidc-mysso` provider.
+======
+----
+CREATE USER alice
+SET HOME DATABASE anotherDb
+SET AUTH PROVIDER 'ldap' { SET ID 'cn=alice,ou=engineering,dc=example,dc=com' }
+SET AUTH 'oidc-mysso1' {SET ID 'alicesUniqueMySso1Id'}
+----
+This will create a user who can authenticate and authorize using `ldap` or `oidc-mysso`. See xref:authentication-authorization/sso-integration.adoc#auth-sso-auth-objects[SSO integration] for more information on setting up an OIDC provider.
+This example also illustrates that the user can have their `HOME` database set even when using an external auth provider.
+======
+
+.Alter a user to remove one of their Auth Objects.
+======
+----
+ALTER USER alice
+REMOVE AUTH 'ldap'
+
+----
+This will prevent the user from being able to authenticate and authorize using `ldap`.
+======
+
+.Alter a user to allow them to authenticate and authorize using username and password
+======
+----
+ALTER USER alice
+SET AUTH 'native' {SET PASSWORD 'changeme' SET PASSWORD CHANGE REQUIRED}
+
+----
+This will allow the user to authenticate and authorize using the specified username and password (in addition to what they are already configured to use).
+======
+
+
+.Configure the database to allow authentication via `ldap` and authorization via the `native` provider.
+======
+
+Set the following database config:
+[source,configuration,role="noheader"]
+----
+dbms.security.authentication_providers=ldap
+dbms.security.authorization_providers=native
+----
+
+Then create a user with an `ldap` Auth Object:
+----
+CREATE USER alice
+SET AUTH PROVIDER 'ldap' { SET ID 'cn=alice,ou=engineering,dc=example,dc=com' }
+----
+
+Then natively grant the `ANALYST` role to the user:
```

Comment:
is this role mentioned elsewhere in this file so readers would know what it is? Or does it need any explanation (I'm just reading the new section so this was the first time I came across it)

Reply:
no, it was introduced here.
I've changed it to the existent `READER` role.

Reply:
This was only updated in the SSO file but missed in the LDAP file :(